### DISCUSSION TITLE
Relationship manager nested pivot relationships

### DISCUSSION THREAD

**Discussion by jason-beach** [filamentphp/filament#7617] [Help] [answered]
We are trying to bring in a title attribute from a pivot with a foreign key of another pivot table.

Models:
- Batch
- OrderItem (pivot) - foreign_keys: order_id, product_id
- BatchOrderItem (pivot) - foreign_keys:  order_item_id, batch_id
- Order - customer_id
- Customer
- Product

We are trying to attach order items to a batch by searching for the customer's name when the attach modal opens up.

I have this setup in the relationship manager:
`protected static string $relationship = 'orderItems';`

Relationships:
Batch Model - 
```shell
public function orderItems(): BelongsToMany
    {
        return $this->belongsToMany(OrderItem::class)
            ->using(BatchOrderItem::class)
            ->withTimestamps();
    }
```

Is there a way we can accomplish this without having to put the customer's name in the order_items table itself? Ideally we'd like to lean on the relationships.

**Comment by danharrin** [comment]
Have you tried customising the AttachAction? 

https://filamentphp.com/docs/3.x/panels/resources/relation-managers#attaching-with-pivot-attributes

For instance, you can chain your own `$action->getRecordSelect()->options([...])`

**Comment by jason-beach** [comment]
@danharrin ,

I'm able to populate the attach select field with the proper information but when I search in the dropdown I get a SQLSTATE[42703]: Undefined column which seems to be looking in the pivot table for a column that isn't there. We don't have information in the pivot table. Could you expound further on your possible idea? We are new to filamentphp...

**Reply by danharrin** [reply, marked as answer]
You probably need to override `getSearchResultsUsing()` as well as `options()`. Check the `Select` docs
